### DISCUSSION TITLE
Droplet Deployment - Time out error

### DISCUSSION THREAD

**Discussion by AnotherITcorp** [Budibase/budibase#4517] [Help] [answered]
Am quite new to budibase and excited to discover this tool but... I not achieve to deploy it  😰  Can you provide me some help ? 

I tried to deploy with 1-click on Digital Ocean just to navigate into the tool and give a try but I can't reach the app on port 10000 after droplet deployment. I tried to update firewall rules, change the port on the config file via SSH... And I still have a connection timed out error

**Comment by lake1984** [comment]
Hi Budibase - same here. Longtime aPaas user. In my case, first time exploring Linux or using CLI beyond a few commands in DOS years ago. I have tried the 1-click option 3 times and each time <ip>:10000 times out. I have spent a few hours trying to learn if there are commands to verify if the app is even running on my server.  I tried various commands including "systemctl status docker" which reports ACTIVE, "docker ps" which returns an empty container list. Also, "docker-compose up" reports the yaml file is invalid since all the variables are not set. As your docs suggest, those with no CLI experience should avoid self-hosting. I appreciate the chance to try. Is it possible a team member could test the 1-click option, confirm it works and report back here to advise if it is deploying for you or if an error was found, etc? I was hoping to just use the 1-click and then put my app experience to work. Thank you

**Reply by AnotherITcorp** [reply]
> Hi Budibase - same here. Longtime aPaas user. In my case, first time exploring Linux or using CLI beyond a few commands in DOS years ago. I have tried the 1-click option 3 times and each time :10000 times out. I have spent a few hours trying to learn if there are commands to verify if the app is even running on my server. I tried various commands including "systemctl status docker" which reports ACTIVE, "docker ps" which returns an empty container list. Also, "docker-compose up" reports the yaml file is invalid since all the variables are not set. As your docs suggest, those with no CLI experience should avoid self-hosting. I appreciate the chance to try. Is it possible a team member could test the 1-click option, confirm it works and report back here to advise if it is deploying for you or if an error was found, etc? I was hoping to just use the 1-click and then put my app experience to work. Thank you

Yep I totally agree that people with no CLI xp and widely no dev xp should not use this kind of tool. I just asking myself on a official procedure seems not working. Moreover I experience the same issues with local docker install...

**Comment by shogunpurple** [comment]
Hi folks - we recently migrated our reverse proxy from envoy to NGINX and it appears it's causing some issues with our digitalocean setup. We will look into this ASAP.

> Yep I totally agree that people with no CLI xp and widely no dev xp should not use this kind of tool. I just asking myself on a official procedure seems not working. Moreover I experience the same issues with local docker install...

The budibase cloud platform exists for this exact purpose - it is not recommended to install and maintain a self hosted system unless you have some experience in doing so. You can sign up for and try the budibase platform in the cloud from the budibase website. Anyone can use budibase, and we provide offerings for both those comfortable self hosting and those who are not.

We will keep this issue updated with the status of the fixes for digitalocean. 

Thanks!

**Reply by AnotherITcorp** [reply]
Do you have some kind of idea on how many time do you need to proceed the changes ? Thx!

**Comment by kpalik** [comment]
i am also in this situaltion, just created new budibase droplet and got timeout using http://<droplet IP>:10000 URL

if you can log into your droplet using SSH you can always use [Budibase CLI Setup](https://docs.budibase.com/self-hosting/self-host/budibase-cli-setup)

in my droplet it was like 7 commands:

apt install npm
npm i -g @budibase/cli
cd /opt
mkdir bbase
cd bbase
budi hosting --init
budi hosting --start

and it worked as promised in docs :) 

i am not pro at linux administration so i don't know if /opt/bbase is good location, or if i need to do some linux hardening (i supose i should), but it is not production enviroment anyway

**Comment by shogunpurple** [comment, marked as answer]
Hey folks - 

The digitalocean droplet issue should now be resolved - bear in mind that once the droplet comes up, it takes around 2-3 minutes to initiate and download all the docker images. Once it's finished, you can access your budibase installation at:

`<droplet-ip>:10000` 

I was just able to successfully spin up a droplet and use it 👍

**Reply by AnotherITcorp** [reply]
Indeed, it seems to be fixed ! Thanks!

**Comment by hendrikjap** [comment]
My recommended setup is to use Budibase CLI. You spin up your Linux instance, SSH into it using Droplet Console, and use NPM to install. It gives you more flexibility and easier to do Caddy reverse proxy.

**Reply by AnotherITcorp** [reply]
Sure, but my concern was to test the Budibase features to show if I choose to deep dive into it and learn. This method was just perfect to do so.

**Reply by shogunpurple** [reply]
@AnotherITcorp if you wanted to just test out budibase features, is there a reason why you didn't use our cloud platform? Just curious. Thanks!

**Reply by AnotherITcorp** [reply]
@shogunpurple cause I just didn't see the option : when I log into Budibase, I have 3 methods - Digital Ocean, Kubernetes and Docker. I do see on your website you have this kind of proposition but I thought it was not yet released.

So either I'm blind (very possible) or it's not as clear.

**Comment by lake1984** [comment]
Same here. After I could not get BB to run on the Droplet, I returned to
the page on your site to try the cloud route. I was only offered the
self-host option.

The updated 1-click worked perfectly last night - thank you. I imported a
csv and created an app in seconds to track activity from a hobby. I look
forward to sharing BB with some clients that should move from Sheets to an
app.

**Reply by AnotherITcorp** [reply]
Feel better on my sharpness, thanks !  😂 

But yes, I agree : when it works, it works incredibly good!
